# Species ID via PCR and Sanger Sequencing

Master Mix Information

[EmeraldAmp® GT PCR Master Mix Cat # RR310A RR310B](https://www.takarabio.com/documents/User%20Manual/RR310A_DS.v1902Da-a_116982.pdf)

#### General Master Mix 25 µl    
|Volume (µl)|	Reagent|
|---|---|
|12.94	|Emerald Master Mix
|0.33	|10µM working stock of For Primer
|0.33	|10µM working stock of Rev Primer
|11.39	|Nuclease Free Water
|25.00	

_**For all markers below Master Mix need to be adapted for EmeraldAmp® GT PCR Master Mix above**_ 
	
#### General PCR profile:
**1 cycle:**
94˚C for 60 sec  

**30 cycles of:**     
94˚C for 30 seconds    
Specific Primer Tm˚C for 30 seconds   
72˚C for 54 sec    

**1 cycle:**   
 72 °C for 5 min.

_**For all markers below need to be adapted for the primer specific Tm**_

30-34 cycles should be fine for all markers


## _Porites_
3 Marker Approaches - https://peerj.com/articles/8550/

### Coral nuclear ITS1-5.8S-ITS2 region (ITS) 

#### Reference
[Forsman et al 2009 ITS](https://bmcecolevol.biomedcentral.com/articles/10.1186/1471-2148-9-45)

#### ITS Primers
ITSZ1 5'-TAA AAG TCG TAA CAA GGT TTC CGT A-3'   
ITSZ2 5'-CCT CCG CTT ATT GAT ATG CTT AAA T-3'
 
expected band size ∼700 bp    
coral nuclear ITS1-5.8S-ITS2 region (ITS) with primers ITSZ1 and ITSZ2 

#### ITS PCR profile: [citation](http://hunterlabhawaii.com/docs/pubs/Forsman%202006%20ICRS.pdf)
1 cycle: 96˚C for 2minutes  
30 cycles of: 96˚C for 10 seconds, 50˚C for 30 seconds, and 70˚C for 4 minutes    
1 cycle: 70 °C for 5 min.

#### ITS Master Mix
I can't find this in the paper. See General PCR profile above

### Coral nuclear histone region spanning H2A to H4 (H2)

#### Reference
[Tisthammer et al PeerJ 2020 nuclear histone region](https://peerj.com/articles/8550/)

#### H2 Primers
zH2AH4f 5′-GTGTACTTGGCTGCYGTRCT-3′    
zH4Fr 5′-GACAACCGAGAATGTCCGGT-3′

expected band size ∼1,500 bp   
coral nuclear histone region spanning H2A to H4 (H2) with novel primers 

#### H2 PCR profile:  
1 cycle: 96 °C for 2 min   
34 cycles: 96 °C for 20 s, 58.5 °C for 20 s, and 72 °C for 90 s   
1 cycle: 72 °C for 5 min.    

#### H2 Master Mix
H2 amplifications (25 µl)    
0.5 µl of DNA template   
0.2 µl of GoTaq® DNA Polymerase (Promega, Madison, WI)    
5 µl of GoTaq® Reaction Buffer    
1.6 µl of 50 mM MgCl2    
2 µl of 10 mM dNTPmix    
1.6 µl of Forward (10 mM) primer 
1.6 µl of Reverse (10 mM) primer   
nuclease-free water to a final volume of 25µl. 


### mtCR mitochondiral control region

#### Reference
[Vollmer & Palumbi 2002 mtCR](https://www.science.org/doi/full/10.1126/science.1069524)

#### mtCR Primers
CRf 5′-GCTTAGACAGGTTGGTTGATTGCCC-3′   
CO3r 5′-CTCCCAAATACATAATTTGAACTAA-3′

expected band size 933+83 bp      
mitochondrial putative control region (933+ bp) plus 83 bp of cytochrome oxidase III was amplified with primers 

#### mtCR PCR profile: 
"Amplifications were obtained with GeneAmp XL PCR kits under normal polymerase chain reaction (PCR) conditions, 30 to 35 cycles, and annealing temperatures of 51° to 54°C." 

General PCR profile above with Tm 53

#### mtCR Master Mix 
I can't find this in the paper. Use General Master Mix 25 µl above

## _Pocillopora_

### mtORF

#### Reference
[Johnston et al PeerJ mtORF](https://peerj.com/articles/4355/)

#### mtORF PCR profile:  
1 cycle: 94 °C for 60 s   
40 cycles: 94 °C for 30 s, 53 °C for 30 s, and 72 °C for 75 s   
1 cycle: 72 °C for 5 min.  

 
#### mtORF Primers 
FatP6.1 5′-TTTGGGSATTCGTTTAGCAG-3′    
RORF 5′-SCCAATATGTTAAACASCATGTCA-3′

#### mtORF Master Mix
mtORF amplifications (15 µl)    
7.5 μL of BioMix (Bioline Ltd., London, UK)    
0.195 μL of forward  primer (10 μM)   
0.195 μL of reverse primer (10 μM)    
0.14 μL of BSA   
1.5 μL of template DNA (5–50 ng)   
5.47 μL of nuclease-free water to 15 μL final volume 

[Notebook Example](https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/mtORF-protocol/)

### PocHistone 3 region

#### Reference
[Johnston et al PeerJ PocHistone](https://peerj.com/articles/4355/)

#### PocHistone PCR profile:  
1 cycle: 94 °C for 60 s   
40 cycles: 94 °C for 30 s, 53 °C for 30 s, and 72 °C for 60 s   
1 cycle: 72 °C for 5 min.   

DNA from individuals of mtORF type 1 (P. meandrina and P. eydouxi) need also to be amplified using novel primers for the histone 3 region 

#### PocHistone Primers
PocHistoneF: 5′-ATTCAGTCTCACTCACTCACTCAC-3′ 
PocHistoneR: 5′-TATCTTCGAACAGACCCACCAAAT-3′

#### PocHistone Master Mix
PocHistone amplifications (15 µl)    
7.5 μL of BioMix (Bioline Ltd., London, UK)    
0.195 μL of forward  primer (10 μM)   
0.195 μL of reverse primer (10 μM)    
0.14 μL of BSA   
1.5 μL of template DNA (5–50 ng)   
5.47 μL of nuclease-free water to 15 μL final volume 


### Additional References: 

[Flot et al 2008](https://www.sciencedirect.com/science/article/pii/S1631069107003812?via%3Dihub)

[Forsman 2003](https://d1wqtxts1xzle7.cloudfront.net/11975375/dissertationfinal-libre.pdf?1390860827=&response-content-disposition=inline%3B+filename%3DPhylogeny_and_Phylogeography_of_Porites.pdf&Expires=1678558521&Signature=NL29RheGNRRhGHLs3N5-NlH7bkLcHtQr~oNFuY~OonbKv1przoAcNbeZOfP9m-wbt28CmNMw~lQoQktTZdjQvGZpm6B2gq7KNSPukM7HR7~5Mn38ZP0pbbm9LOLvOYU5GKKOlLQyO7QnumgbkqlQPGeBr6QsBRu7-JzdjZvgaMGzm8uz4i55L5j6wIR59Cehl~w3KoAZqfJg-lE5JQMS7CKWVCRzv9i6-B~1vn0I~EvY0aKxVElwsXk0cfVN5LCj5dqC-zNfYPtTpZ7E-oABABis5XpOeuf7Oz2vt03vJ47i1LPC9PbyVN7AaqIwbSjQfLk0k1SHly-nehPI2OojUg__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA)